test/topology_experimental_raft/test_tombstone_gc.py

```diff
@@ -44,3 +50,137 @@ async def test_default_tombstone_gc_does_not_override(manager: ManagerClient, rf
+        return metadata.get("tombstone")
+
+    async def list_tombstones(tombstone_mark, host):
+        tombstones = {}
+        for tbl in ("tables", "columns"):
+            res = list(cql.execute(f"SELECT * FROM MUTATION_FRAGMENTS(system_schema.{tbl})", host=host))
+            tombstones[tbl] = []
+            for row in res:
+                tombstone = get_tombstone(row)
+                if tombstone and datetime.fromtimestamp(float(tombstone["timestamp"])/1_000_000, timezone.utc) < tombstone_mark:
+                    tombstones[tbl].append(tombstone)
+        return tombstones
+
+    async def tombstone_gc_completed(tombstone_mark):
+        # flush and compact the keyspace
+        await asyncio.gather(*[manager.api.keyspace_flush(srv.ip_addr, "system_schema")
+                               for srv in servers])
+        await asyncio.gather(*[manager.api.keyspace_compaction(srv.ip_addr, "system_schema")
+                               for srv in servers])
+
+        # check the remanining tombstones
+        tombstones_count_total = 0
+        tombstones_per_host = await asyncio.gather(*[list_tombstones(tombstone_mark, host)
+                                                     for host in hosts])
+        for tombstones in tombstones_per_host:
+            for tbl in tombstones.keys():
+                tombstones_remaining = tombstones[tbl]
+                tombstones_count = len(tombstones_remaining)
+                tombstones_count_total += tombstones_count
+                logger.info(f"{tbl} tombstones remaining: {tombstones_count}")
+        if tombstones_count_total != 0:
+            return None
+        return True
+
+    # should usually run much faster than 30s, but left some margin to avoid flakiness
+    async def verify_tombstone_gc(tombstone_mark, timeout=30):
+        # wait for 1 sec to let the current tombstones fully expire
+        await asyncio.sleep(1)
+
+        # perform a single change to generate a new state_id and set the previous tombstones to expire
+        # (this is needed because we deduct 1s of the tombstone expiration time to account for the changes coming
+        # in the same second)
+        await alter_system_schema(table_count=1)
+
+        await wait_for(partial(tombstone_gc_completed, tombstone_mark), time.time() + timeout)
+
+    async with new_test_keyspace(cql, "with replication = { 'class': 'NetworkTopologyStrategy', 'replication_factor': 2}") as keyspace:
+        await alter_system_schema(keyspace)
+        tombstone_mark = datetime.now(timezone.utc)
+        await verify_tombstone_gc(tombstone_mark)
+
+        # shut down one server
+        down_server = servers.pop()
+        down_host = hosts.pop()
+        await manager.server_stop_gracefully(down_server.server_id)
+        await asyncio.gather(*[manager.server_not_sees_other_server(srv.ip_addr, down_server.ip_addr)
+                               for srv in servers])
+
+        await alter_system_schema(keyspace)
```

Comment:
Added a helper to switching the agreement wait off (and set it back as a context handler) ... but it looks that the test still takes the same time, so it seems that the delay didn't apply (the test was always taking about 1 min., so in fact if the 20 sec on every change would apply with the 5 creates / alters / drops the delay would be maybe at least 1:40 min or even 5 min if it would apply to all ops)

Reply:
So the driver is probably ignoring down nodes and not waiting for them